bftengine/include/bftengine/ReplicaConfig.hpp

```diff
@@ -283,6 +283,26 @@ class ReplicaConfig : public concord::serialize::SerializableFactory<ReplicaConf
                concord::crypto::SignatureAlgorithm::EdDSA,
                "A flag to specify the operator message signing algorithm. It is defaulted to use EDDSA algo.");
 
+  CONFIG_PARAM(enableSocketKeepAlive, bool, true, "A flag to indicate if TCP socket keep-alive is enabled.");
+
+  CONFIG_PARAM(socketKeepAliveIdleTime,
+               uint32_t,
+               600,
+               "In case the TCP socket keep-alive is enabled, define how many seconds to wait on an idle socket to "
+               "start sending keep-alive packets.");
+
+  CONFIG_PARAM(socketKeepAliveInterval,
+               uint32_t,
+               2,
+               "In case the TCP socket keep-alive is enabled, define how frequently (in seconds) keep-alive packets "
+               "should be sent.");
+
+  CONFIG_PARAM(socketKeepAliveProbesNum,
+               uint32_t,
+               10,
+               "In case the TCP socket keep-alive is enabled, define how many packet should be lost until the socket "
```

Comment:
so after 10 minutes we will wait 20 seconds until we fail?

Reply:
Yes